python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -536,3 +563,150 @@ def to_arrow(self):
                 arrow_polygons,
             ],
         )
+
+    def contains(self, other, align=True):
+        """Compute from a set of points and a set of polygons which points fall
+        within each polygon. Note that `polygons_(x,y)` must be specified as
+        closed polygons: the first and last coordinate of each polygon must be
+        the same.
+
+        This implements `.contains_properly`, which shares a large
+        space of correct cases with `GeoPandas.contains` but they do not
+        produce identical results. In the future we will use intersection
+        testing to match .contains behavior.
+
+        Parameters
+        ----------
+        test_points_x
+            x-coordinate of test points
+        test_points_y
+            y-coordinate of test points
+        poly_offsets
+            beginning index of the first ring in each polygon
+        poly_ring_offsets
+            beginning index of the first point in each ring
+        poly_points_x
+            x closed-coordinate of polygon points
+        poly_points_y
+            y closed-coordinate of polygon points
+
+        Examples
+        --------
+
+        Test if a polygon is inside another polygon:
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0.5, 0.5)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
```

Comment:
For these examples, can't you create the GeoSeries directly on the device to avoid the calls to `from_geopandas`?

Reply:
It is possible, but uglier than using GeoPandas. GeoPandas is intended to be used as the I/O layer for cuSpatial right now, the only other I/O other than directly manual is to use a Shapefile. I'll look into it.

Reply:
Ah, so we don't yet have a GeoSeries constructor from host data?

```
polygon = cuspatial.GeoSeries([Point(0.5, 0.5)])
```

Reply:
No, though it would be trivial to do this and use GeoPandas under the hood. A good idea!

Reply:
Create `cuspatial.GeoSeries` objects directly from Shapely arrays.

Reply:
Actually maybe we can already do this.

Reply:
Good idea, done.

Reply:
I was mistaken about it being ugly!